### PR TITLE
Bumping main version to opensearch core 2.1.0

### LINKED ISSUE #385: Release version 2.0.0 (GA)
This is a component issue for 2.0.0 (GA).
Coming from [opensearch-build 2086](https://github.com/opensearch-project/opensearch-build/issues/2086). Please follow the following checklist.
__Please refer to the DATES in that post__.

<details><summary>How to use this issue</summary>
<p>

## This Component Release Issue

This issue captures the state of the OpenSearch release, on component/plugin level; its assignee is responsible for driving the release. Please contact them or @mention them on this issue for help. 
Any release related work can be linked to this issue or added as comments to create visiblity into the release status.

## Release Steps

There are several steps to the release process; these steps are completed as the whole component release and components that are behind present risk to the release. The component owner resolves the tasks in this issue and communicate with the overall release owner to make sure each component are moving along as expected.

Steps have completion dates for coordinating efforts between the components of a release; components can start as soon as they are ready far in advance of a future release. The most current set of dates is on the overall release issue linked at the top of this issue.

## The Overall Release Issue

Linked at the top of this issue, the overall release issue captures the state of the entire OpenSearch release including references to this issue, the release owner which is the assignee is responsible for communicating the release status broadly.  Please contact them or @mention them on that issue for help.

## What should I do if my plugin isn't making any changes?

If including changes in this release, increment the version on `2.0` branch to `2.0.0` for Min/Core, and `2.0.0.0` for components. Otherwise, keep the version number unchanged for both.

</p>
</details>

### Preparation

- [x] Assign this issue to a release owner.
- [x] Finalize scope and feature set and update [the Public Roadmap](https://github.com/orgs/opensearch-project/projects/1).
- [x] All the tasks in this issue have been reviewed by the release owner.
- [x] Create, update, triage and label all features and issues targeted for this release with `v2.0.0`.

### CI/CD

- [x] All code changes for `2.0.0` are complete.
- [x] If this repo is core repo, use `2.0.0` version for your component; if this repo is plugin repo, use `2.0.0.0` version for your component. Qualifier should be empty `''` see this issue for details: https://github.com/opensearch-project/opensearch-build/issues/2078. 
- [x] Ensure working and passing CI.
- [x] Check that this repo is included in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.0.0).

### Pre-Release

- [x] Update to the `2.0` release branch in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.0.0).
- [x] Increment the version on the parent branch to the next development iteration.
- [x] Gather, review and publish release notes following the [rules](https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md) and back port it to the release branch.[git-release-notes](https://github.com/ariatemplates/git-release-notes) may be used to generate release notes from your commit history.
- [x] Confirm that all changes for `2.0.0` have been merged.

### Release Testing

- [x] Find/fix bugs using latest tarball and docker image provided in parent release issue and update the release notes if necessary.
- [x] Code Complete: Test within the distribution, ensuring integration, backwards compatibility, and performance tests pass.
- [x] Sanity Testing: Sanity testing *and* fixing of critical issues found.
- [x] File issues for all intermittent test failures.

### Release

- [x] Complete [documentation](https://github.com/opensearch-project/documentation-website).
- [x] Verify all issued labeled for this release are closed or labelled for the next release.

### Post Release

- [x] Prepare for an eventual security fix development iteration by incrementing the version on the release branch to the next eventual patch version.
- [x] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) of the next eventual security patch version.
- [x] [Suggest improvements](https://github.com/opensearch-project/opensearch-build/issues/new) to [this template](https://github.com/opensearch-project/opensearch-build/blob/main/meta/templates/releases/release_template.md).
- [x] Conduct a retrospective, and publish its results.

### PR DESCRIPTION
**Pull request by martin-gaievski**
Signed-off-by: Martin Gaievski <gaievski@amazon.com>

### Description
Upgrade core OpenSearch version to 2.1.0 after 2.0.0 GA release. 

### Issues Resolved
#385 
 
### Check List
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -44,6 +44,7 @@ apply from: 'gradle/formatting.gradle'
 apply plugin: 'opensearch.opensearchplugin'
 apply plugin: 'opensearch.rest-test'
 apply plugin: 'opensearch.pluginzip'
+apply plugin: 'opensearch.repositories'
```
Comment (navneet1v) [inline build.gradle:47]: Why we need this?

Comment (martin-gaievski) [reply build.gradle:47]: it's required to get some dependencies for Core OpenSearch, for instance for us it was failing without this plugin on getting proper lucene version, please see details here https://github.com/opensearch-project/OpenSearch/issues/3455

**Review by navneet1v — COMMENTED** [review]
Please check the comment in build.gradle file, around why we need 

`apply plugin: 'opensearch.repositories'`

File: build.gradle (hunk 1 above)
Comment (navneet1v) [reply build.gradle:47]: Is this is the best way to get the lucene, what about the snapshot provided in the github issue?

`maven { url "https://d1nvenhzbhpy0q.cloudfront.net/snapshots/lucene/" }`

Comment (martin-gaievski) [reply build.gradle:47]: both approaches recommended by Core team, my guess they are equivalent

**Review by navneet1v — DISMISSED** [review]

File: .github/workflows/backwards_compatibility_tests_workflow.yml [hunk 2]
```diff
@@ -46,8 +46,8 @@ jobs:
     strategy:
       matrix:
         java: [ 11, 17 ]
-        bwc_version: [ "1.3.2" ]
-        opensearch_version: [ "2.0.0-rc1-SNAPSHOT", "2.0.0-SNAPSHOT" ]
+        bwc_version: [ "2.0.0" ]
+        opensearch_version: [ "2.1.0-SNAPSHOT" ]
```
Comment (naveentatikonda) [inline .github/workflows/backwards_compatibility_tests_workflow.yml:50]: For rolling upgrade tests, we will upgrade from the last minor version of previous major version to all minor versions of current major version.
Reference - https://quip-amazon.com/lzkEA6OVlmjv#temp:C:SFc6fa8c229d21a652b821c326b3

```
bwc_version: [ "1.3.2" ]
opensearch_version: [ "2.0.0-SNAPSHOT", "2.1.0-SNAPSHOT" ]

File: qa/restart-upgrade/build.gradle [hunk 3]
```diff
@@ -51,7 +51,7 @@ testClusters {
         }
 
         // Skip test if version is 1.0, 1.1, 1.2 or 1.3 as they are not supported in those versions
-        if (knn_bwc_version.startsWith("1.")) {
+        if (knn_bwc_version.startsWith("1.") || knn_bwc_version.startsWith("2.0")) {
```
Comment (VijayanB) [inline qa/restart-upgrade/build.gradle:54]: Why not replace "1." with "2."?

File: build.gradle (hunk 1 above)
Comment (martin-gaievski) [reply build.gradle:47]: This post provides good explanation - https://github.com/opensearch-project/OpenSearch/issues/2500#issuecomment-1072046551

File: qa/restart-upgrade/build.gradle (hunk 3 above)
Comment (naveentatikonda) [reply qa/restart-upgrade/build.gradle:54]: We are also running BWC Tests from 1.x versions to 2.x versions. So, we need to add that condition

Comment (martin-gaievski) [reply qa/restart-upgrade/build.gradle:54]: My understanding is we need both. opensearch_version defines current version, knn_bwc_version defines the version we want to be compatible with, so setting those to let's say 2.1.0 and 1.3.0 we testing if 2.1.0 knn is backward compatible with 1.3.0.
We are checking such compatibility for all versions defined [here](https://github.com/opensearch-project/k-NN/blob/main/.github/workflows/backwards_compatibility_tests_workflow.yml#L17) https://github.com/opensearch-project/k-NN/actions/runs/2398778347

File: .github/workflows/backwards_compatibility_tests_workflow.yml (hunk 2 above)
Comment (martin-gaievski) [reply .github/workflows/backwards_compatibility_tests_workflow.yml:50]: ack

**Review by naveentatikonda — APPROVED** [review]

**Review by navneet1v — APPROVED** [review]

File: qa/restart-upgrade/build.gradle (hunk 3 above)
Comment (martin-gaievski) [reply qa/restart-upgrade/build.gradle:54]: @VijayanB does it make sense? Actually we've added rolling upgrade bwc for both 1.3.2 and 2.0.0

**Review by VijayanB — APPROVED** [review]

**Pull request merged**